### PR TITLE
Pgsearch(search on projects)

### LINKED ISSUE #21: Search Engine
Currently we are using google custom search which doesn't do a very good job. The search should search over project names and descriptions.

### PR DESCRIPTION
**Pull request by Amansingla97**
Fixes #21 

Implemented full text search with pg_search.Removed sql specific code with minor changes in UI.

### CODE REVIEW

File: app/models/project.rb [hunk 1]
```diff
@@ -12,10 +13,21 @@ class Project < ApplicationRecord
...
+
   self.per_page = 8
 
   acts_as_commontable
   # after_commit :send_mail, on: :create
+  def self.search(query)
+    if query.present?
+      Project.text_search(query)
+    else
+      Project.all
```
Comment (tachyons) [inline app/models/project.rb]: Can we have default limit for this as well

Comment (Amansingla97) [reply app/models/project.rb]: what do you think default limit should be set to ? According to me 50 is good enough.

**Comment by satu0king** [comment]
Looks good to me, will merge once @tachyons approves.

File: config/initializers/search_paginate_renderer.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      link_wrapper = link(text, page, link_options.merge(class: 'page-link'))
+      tag :li, link_wrapper, class: 'page-item'
+    else
+      span_wrapper = tag(:span, text, class: 'page-link')
+      tag :li, span_wrapper, class: 'page-item disabled'
+    end
+  end
+
+  def gap
+    tag :li, tag(:i, ELLIPSIS, class: 'page-link'), class: 'page-item disabled'
```
Comment (satu0king) [inline config/initializers/search_paginate_renderer.rb]: @Amansingla97 It says this: `ELLIPSIS` is not defined and hence giving an error in production

**Pull request merged**